src/Interfaces.jl

```diff
@@ -94,22 +94,57 @@ function methodparts(IT, x::Expr)
+function requiredreturn(IT, nm, args, shouldthrow, RT_sym, __RT__)
     return quote
-        check = $(requiredmethod(T, nm, args, shouldthrow))
+        $(requiredmethod(IT, nm, args, shouldthrow))
         $RT_sym = Interfaces.returntype($nm, $args)
         # @show $RT_sym, $nm, $args, Interfaces.isinterfacetype($__RT__)
-        check |= Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
-        check || ($shouldthrow && Interfaces.invalidreturntype($nm, $args, $RT_sym, $__RT__))
+        check = Interfaces.isinterfacetype($__RT__) ?  Interfaces.implements($RT_sym, $__RT__) : $RT_sym <: $__RT__
+        check || ($shouldthrow && Interfaces.invalidreturntype(debuglvl, $IT, $nm, $args, $RT_sym, $__RT__))
+        check
     end
 end
 
-@noinline missingmethod(IT, f, args, mods) = throw(InterfaceImplementationError("missing `$IT` interface method definition: `$(Expr(:call, f, unconvertargs(args)...))`, in module(s): `$mods`"))
-@noinline invalidreturntype(IT, f, args, RT1, RT2) = throw(InterfaceImplementationError("invalid return type for `$IT` interface method definition: `$(Expr(:call, f, unconvertargs(args)...))`; inferred $RT1, required $RT2"))
-@noinline subtypingrequired(IT, T) = throw(InterfaceImplementationError("interface `$IT` requires implementing types to subtype, like: `struct $T <: $IT`"))
-@noinline atleastonerequired(IT, expr) = throw(InterfaceImplementationError("for `$IT` interface, one of the following method definitions is required: `$expr`"))
+@noinline function missingmethod(debuglvl, IT, f, args, mods)
+    msg = "missing `$IT` interface method definition: `$(Interfaces.methodsig(f, args))`, in module(s): `$mods`"
+    debuglvl == :error ? print_error(msg) : debuglvl == :warn ? print_warn(msg) : throw(InterfaceImplementationError(msg))
+end
+
+@noinline function invalidreturntype(debuglvl, IT, f, args, RT1, RT2)
+    msg = "invalid return type for `$IT` interface method definition: `$(methodsig(f, args))`; inferred $RT1, required $RT2"
+    debuglvl == :error ? print_error(msg) : debuglvl == :warn ? print_warn(msg) : throw(InterfaceImplementationError(msg))
+end
+
+@noinline function subtypingrequired(debuglvl, IT, T)
+    msg = "interface `$IT` requires implementing types to subtype, like: `struct $T <: $IT`"
+    debuglvl == :error ? print_error(msg) : debuglvl == :warn ? print_warn(msg) : throw(InterfaceImplementationError(msg))
+end
+
+@noinline function atleastonerequired(debuglvl, IT, expr)
+    msg = "for `$IT` interface, one of the following method definitions is required: `$expr`"
+    debuglvl == :error ? print_error(msg) : debuglvl == :warn ? print_warn(msg) : throw(InterfaceImplementationError(msg))
+end
+
+function print_error(msg)
+    printstyled("[ InterfaceImplementationError: "; color=Base.error_color())
+    println(msg)
+end
+
+function print_warn(msg)
+    printstyled("[ @optional definition missing: "; color=Base.warn_color())
+    println(msg)
```

Comment:
i'm not at all committed to how these currently display, they're just something to get us started